__tests__/task.test.js

```diff
@@ -1,153 +1,146 @@
-  });
-
-  describe('read', () => {
-    it('should return 200', async () => {
-      const res = await server.inject({
-        method: 'get',
-        url: '/tasks',
-      });
-      expect(res.statusCode).toBe(200);
+    existingStatus = await app.objection.models.status.query().insert({
+      name: 'test status',
     });
   });
 
   describe('create', () => {
     describe('when using valid data', () => {
       it('should return 302', async () => {
-        const res = await server.inject({
+        const name = 'test task';
+        const description = 'test description';
+        const { statusCode } = await app.inject({
           method: 'post',
-          url: '/tasks',
+          url: '/task',
           body: {
-            name: 'test',
-            description: 'test',
-            status_id: status.id,
-            creator_id: user.id,
+            name,
+            description,
+            status_id: existingStatus.id,
+            creator_id: existingUser.id,
             assigned_id: null,
           },
         });
-        expect(res.statusCode).toBe(302);
+        expect(statusCode).toBe(302);
+
+        const tasks = await app.objection.models.task.query();
+        expect(tasks).toHaveLength(1);
+        expect(tasks[0]).toMatchObject({
+          name,
+          description,
+          statusId: existingStatus.id,
+          creatorId: existingUser.id,
+        });
       });
     });
-    describe.skip('when using invalid data', () => {
+
+    describe('when using invalid data', () => {
       [
-        { name: 'name', body: { description: 'test', status_id: status.id, creator_id: user.id } },
-        { name: 'description', body: { name: 'test', status_id: status.id, creator_id: user.id } },
-        { name: 'status', body: { name: 'test', description: 'test', creator_id: user.id } },
-        { name: 'creator', body: { name: 'test', description: 'test', status_id: status.id } },
-      ].forEach(({ name, body }) => {
+        { name: 'name', data: () => ({ description: 'test', status_id: existingStatus.id, creator_id: existingUser.id }) },
+        { name: 'description', data: () => ({ name: 'test', status_id: existingStatus.id, creator_id: existingUser.id }) },
+        { name: 'status', data: () => ({ name: 'test', description: 'test', creator_id: existingUser.id }) },
+        { name: 'creator', data: () => ({ name: 'test', description: 'test', status_id: existingStatus.id }) },
+      ].forEach(({ name, data }) => {
         it(`should return 400 when missing required field ${name}`, async () => {
-          const res = await server.inject({
+          const { statusCode } = await app.inject({
             method: 'post',
-            url: '/tasks',
-            body,
+            url: '/task',
+            body: data(),
           });
-          expect(res.statusCode).toBe(400);
+          expect(statusCode).toBe(400);
+          const tasks = await app.objection.models.task.query();
```

Comment:
Again about abscence of filtering. It appears test knows about it environment. Why should we conform envrionment to test?

Reply:
that sounds like something bad :)

there is term that tests should not know about their env, otherwise they are considered possibly affected by 'the env'.

but that is a bit of genalization, in fact by 'env' usually mean 'other tests and things which could do unexpected mutations or other effects on current test'

so for example when using single state and > 1 tests running at the same time, they can mess each other...

in our case things much simpler - we have single test running at a time ( jest runs tests in spec in sequence) on its own state ( since database inmemory ) in a single process  ( jest runs tests in parallel with separate process/workers )

so each test as atomic as possible, so all possible state which could be - caused by test, so it should know that to compare actual state with expecting

Reply:
You are totally right. I did poor job explaining myself. When I read the test, I don't usually look at the setup. And why should I? A filter in query will just drops all questions about absence of filtering.

Reply:
lets discuss that separately in chat or voice :)